### ISSUE TITLE
Build step to generate stubs from idl fails.

### ISSUE THREAD

**Issue by Gavin-Williams** [MicrosoftDocs/windows-uwp#793]
After editing the idl I try to build but get …
C1083 Cannot open include file: Class.h: // doesn't exist
C1083	Cannot open include file: 'Class.g.h': No such file or directory (compiling source file Generated Files\module.g.cpp)	
PRI252	0xdef00071 - File D:\OneDrive\Projects\TestSoln\Debug\BankAccountWRC\BankaAccountWRC.pri not found.	

Edit: So it does generate the cpp and h files but they are hidden, and not in the project. And then when I drag them into the project root, I get an error (Failed to drop file … onto project BankAccountWRC. They are not in the project. Then when I try to add them to the project, I get another error 'A file, filter or folder already exists at the current scope in BankAccountWRC or access was denied for the desired operation.

Edit: Restarting VS I see them in the project. But they are not children to the idl like the templates.

Edit: I then can't build with 2 errors:
E1696 cannot open source file pch.h
LNK1104 cannot open file 'BankAccountWRC.def'

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cbf411b4-9c85-5831-c425-a009d47811e7
* Version Independent ID: 3d0dc3b5-a7da-2755-0a9e-599f39a51acd
* Content: [Author events in C++/WinRT - UWP app developer](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/author-events)
* Content Source: [windows-apps-src/cpp-and-winrt-apis/author-events.md](https://github.com/MicrosoftDocs/windows-uwp/blob/docs/windows-apps-src/cpp-and-winrt-apis/author-events.md)
* Product: **windows**
* GitHub Login: @stevewhims
* Microsoft Alias: **stwhi**

**Comment by stevewhims** [comment]
Hi, @Gavin-Williams, thanks for describing these issues, and I'm sorry for the trouble. Yes, after editing the idl, you will get build errors, as the topic says. The topic doesn't explicitly say exactly what the errors will be, though, which is not great. So, I've just edited the topic to add that detail, because I can see how it's nice to know that the errors you're seeing are the expected ones, and not some different ones. I expect that change to be live within 24 hours.

The first 2 errors you're seeing are exactly the errors to be expected at that stage in the walkthrough. But I can't account for the third one. Your folder structure is different than the walkthrough calls for. It looks as if you first created an empty solution named TestSoln, and then added a Runtime Component project into that named BankAccountWRC. But I haven't verified that that works with these steps. Tbh I don't see why it wouldn't, but then again I haven't verified it so I can't be sure. If you stick to the steps as written then that'll maximize the chances of success because I've verified that these steps work (on the recommended OS and toolchain). There's also a curious typo in your error message (BankaAccountWRC), which I can't account for because your parent folder of that doesn't contain that typo, and presumably you copy-pasted the error from VS rather than typing it. So that's strange and like I say I can't explain it to myself but it's certainly not correct.

I can't account for why your cpp and h are hidden. I don't see that, myself, if I follow the steps. Did you do the step of renaming the idl file? If so then you'll notice that the cpp and h files become renamed, too, and they remain included in the project and nested under the idl. If anything goes wrong at this stage then I'd advise stopping there and reporting. Going further might compound the problem, so I wouldn't. For example, the steps do not call for dragging the files onto the project root. They only call for copying them in the file system. Of course, my walkthroughs are not always 100% correct (either because I've made a mistake, or the product has changed since I published the topic). So I am always interested to hear when people run into issues when following my steps, and I'll investigate.

Anyway, the first thing that I see that's wrong is that you're getting that 3rd build error immediately after renaming, pasting in the IDL, and building. If you wouldn't mind, here's what I would suggest. Always be sure you have the recommended version of [Visual Studio and the VSIX](https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/intro-to-using-cpp-with-winrt#visual-studio-support-for-cwinrt-and-the-vsix). Then begin the walkthrough again and double-check that you're following the steps just so. Take precautions so as to minimize the possibility of errors; for example prefer to copy-paste rather than type. For example, when I'm following these steps I double-click a name in the topic and copy-paste that into the relevant UI in Visual Studio in order to eliminate any chance of me making a typographical error like I might if I were typing. And if your error about the PRI file persists then I'd be interested in knowing that and I'll see whether I can help figure out why that's happening with your setup. As I say, it's not happening with my setup with these steps.

Thanks again! ~Steve
